saleor/webhook/observability/worker.py

```diff
@@ -0,0 +1,190 @@
+import atexit
+import logging
+import os
+from queue import Empty, Full, Queue
+from threading import Lock, Thread
+from time import monotonic
+from typing import Any, Callable, Dict, List, Optional
+
+from django.conf import settings
+
+from .buffers import get_buffer
+from .tracing import opentracing_trace
+
+_worker: Optional["BackgroundWorker"] = None
+_TERMINATOR = object()
+FLUSH_TIMEOUT = 2.0
+
+logger = logging.getLogger(__name__)
+
+
+def shutdown(timeout=FLUSH_TIMEOUT) -> bool:
+    global _worker
+    if _worker:
+        logger.debug("[Observability] Background worker shutdown")
+        success = _worker.flush(timeout)
+        _worker = None
+        return success
+    return False
+
+
+def init():
+    global _worker
+    if _worker is None:
+        _worker = BackgroundWorker(
+            batch_size=settings.OBSERVABILITY_BUFFER_BATCH_SIZE,
+            timeout=settings.OBSERVABILITY_REPORT_PERIOD.total_seconds(),
+        )
+        atexit.register(shutdown)
+
+
+def put_event(buffer_name: str, generate_payload: Callable[[], Any]) -> bool:
+    if _worker:
+        return _worker.submit_event(buffer_name, generate_payload)
+    logger.warning("[Observability] Worker not initialized, event dropped")
+    return False
+
+
+def queue_join(timeout: float):
+    if _worker:
+        return _worker.queue_join(timeout)
+    return False
+
+
+def buffer_put_multi_key_events(events: Dict[str, List]) -> bool:
+    dropped_events = get_buffer("").put_multi_key_events(events)
+    all_events_delivered = True
+    for buffer_name, events_count in dropped_events.items():
+        if events_count:
+            all_events_delivered = False
+            logging.warning(
+                "[Observability] Buffer %s full, %s event(s) dropped",
+                buffer_name,
+                events_count,
+            )
+    return all_events_delivered
+
+
+class BackgroundWorker:
+    def __init__(self, batch_size: int, timeout: float):
+        self._batch_size = max(batch_size, 1)
```

Comment:
Why not just `self._batch_size = batch_size`? Seems rather odd to silently change the value

Reply:
I agree. Fixing that 👍